substrate/runtime/contract/src/lib.rs

```diff
@@ -126,13 +136,42 @@ impl<'a, T: Ext + 'a> Runtime<'a, T> {
 	fn ext_mut(&mut self) -> &mut T {
 		self.ext
 	}
+	/// Save a data buffer as a result of the execution.
```

Comment:
care should be taken to ensure that return data buffers never exist outside of the runtimes, otherwise there's the possibility of a quadratic gas attack.